File: srcs/containers/django/src/pong/static/js/module/3d-pong/assets/Paddle.js
import * as THREE from '../three.module.js';

class Paddle {
	constructor (scene, distanceFromCenter, width = 2) {
	this.geometry = new THREE.BoxGeometry(0.3, 0.8, width);
	this.material = new THREE.MeshStandardMaterial({
		color: 0xc1d1db,
		roughness: 0.5,
		metalness: 0.5
	});
    this.dz = 0;
	this.mesh = new THREE.Mesh(this.geometry, this.material);
	this.mesh.position.set(distanceFromCenter, (0.8 / 2) + 0.5 , 0);
	scene.add(this.mesh);
	this.speed = 0.15;
	}
    
	movePaddle(direction, field ) {
        this.dz = direction * this.speed;
        this.mesh.position.z += this.dz;
   
        const halfPaddle = this.geometry.parameters.depth / 2;
        const halfField = field.geometry.parameters.depth / 2;
        const topPaddle = this.mesh.position.z + halfPaddle;
        const bottomPaddle = this.mesh.position.z - halfPaddle;
    
        if (bottomPaddle < -halfField) {
            this.mesh.position.z = halfPaddle - halfField;
        } else if (topPaddle > halfField) {
            this.mesh.position.z = halfField - halfPaddle;
        }
	};

	get position() {
		return this.mesh.position;
	}
}

export { Paddle };

File: srcs/containers/django/src/pong/paddle.py
class Paddle:
    def __init__(self, court_width, left=True, width=2):
        self.width = width
        self.depth = 0.8  # Depth of the paddle (along the z-axis)
        self.position = {'x': 0, 'z': 0}
        self.speed = 0.2

        offset = court_width / 15
        distance_from_center = (court_width / 2) - offset
        if left:
            distance_from_center *= -1
        self.position['x'] = distance_from_center

    def movePaddle(self, direction, court_depth):
        self.position['z'] += direction * self.speed

        half_paddle = self.depth / 2
        half_court = court_depth / 2
        top_paddle = self.position['z'] + half_paddle
        bottom_paddle = self.position['z'] - half_paddle

        if bottom_paddle < -half_court:
            self.position['z'] = half_paddle - half_court
        elif top_paddle > half_court:
            self.position['z'] = half_court - half_paddle

    @property
    def x(self):
        return self.position['x']

    @property
    def z(self):
        return self.position['z']
